.circleci/config.yml

```diff
@@ -1,50 +1,84 @@
 version: 2.1
 
+defaults: &defaults
+  parameters:
+    version:
+      type: string
+      default: "latest"
+  docker:
+    - image: "circleci/golang:<< parameters.version >>"
+  environment:
+    GO111MODULE: "on"
+
 jobs:
+  # Run automated tests
   test:
-    parameters:
-      version:
-        type: string
-        default: "1.16.3"
-      golint:
-        type: boolean
-        default: true
-    docker:
-      - image: "circleci/golang:<< parameters.version >>"
-    environment:
-      GO111MODULE: "on"
+    <<: *defaults
     steps:
       - checkout
       - run:
           name: Print the go version
-          command: >
+          command: |
             go version
       - run:
           name: Fetch dependencies
-          command: >
+          command: |
             go get -v ./...
+            GO111MODULE=off go get -u github.com/mgechev/revive
       - run:
           name: Run gofmt
-          command: >
+          command: |
             diff -u <(echo -n) <(gofmt -d -e .)
       - run:
           name: Run go vet
-          command: >
+          command: |
             go vet -v ./...
       - run:
           name: Run revive
           command: |
-            GO111MODULE=off go get -u github.com/mgechev/revive
             revive ./...
       - run:
           name: Run go test (+ race detector)
-          command: >
+          command: |
             go test -v -race ./...
 
+  # Whenever we create a GitHub release, build binaries for each OS, and upload them into the release
+  release:
+    <<: *defaults
+    steps:
+      - checkout
+      - run:
+          name: Fetch dependencies
+          command: |
+            go get -v ./...
+            go get github.com/github-release/github-release
+
+      - run:
+          name: Build the binaries for each OS
+          command: |
+            CGO_ENABLED=0 GOOS=darwin GOARCH=amd64 go build -o bin/walletnode-darwin-amd64 -ldflags "-X github.com/pastelnetwork/go-commons/version.version=$CIRCLE_TAG -extldflags '-static'" .
```

Comment:
Disabled cgo seems will not work after integration of https://github.com/pastelnetwork/go-raptorq